### ISSUE TITLE
Document EnableSensitiveDataLogging

### ISSUE THREAD

**Issue by divega** [dotnet/EntityFramework.Docs#673]
_From @Wain123 on June 22, 2017 11:7_

I have an ILogger with the following function:
```
public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter) {
  Console.WriteLine(formatter(state, exception));
}
```
DbCommands get logged and the CommandText looks correct, but all the parameter values are simply '?'. Here's a sample output from the Logger:
```
Executed DbCommand (78ms) [Parameters=[p0='?', p1='?' (Size = 1), p2='?', p3='?' (Size = 6), p4='?', p5='?' (Size = 1), p6='?', p7='?', p8='?' (Size = 6)], CommandType='Text', CommandTimeout='0']
INSERT INTO FRM_LAYOUT_FILTER (SEQNR, BEZEICHNUNG, CREATETIME, CREATEUSER, DEFAULT_FILTER, FORM_NAME, MODIFY_NR, MODIFYTIME, MODIFYUSER)
VALUES (:p0, :p1, :p2, :p3, :p4, :p5, :p6, :p7, :p8)
```

_Copied from original issue: aspnet/EntityFrameworkCore#8933_

**Comment by divega** [comment]
_From @smitpatel on June 22, 2017 14:21_

By default EF doesn't put data in logs as the data may contain sensitive information. To get those parameter values, you need to chain `EnableSensitiveDataLogging()` after your `UseSqlServer` (or equivalent method)
e.g.
`builder.UseSqlServer("connectionstring").EnableSensitiveDataLogging()`

**Comment by divega** [comment]
_From @jayoungers on August 23, 2017 17:55_

How is that not mentioned here?

https://docs.microsoft.com/en-us/ef/core/miscellaneous/logging

**Comment by divega** [comment]
_From @smitpatel on August 23, 2017 21:26_

@divega For doc issue

**Comment by divega** [comment]
_From @ajcvickers on April 10, 2018 18:57_

@divega Can this be moved out of the 2.1 milestone. (We are trying to make 2.1 only include issues that we must fix for 2.1)
